### PR TITLE
NOT MERGE READY: Style doc site

### PR DESCRIPTION
**Pull request by ameseee**
‼️Updated after Tanner's comments were made
#### What does this PR do?

- Styles header to match Terminal and fit a bit better with that design
- Adjust footer responsiveness issue
- Add favicon
- Adjust left-side nav bar, some small styling things as well as add sub-nav items
- Add in rouge, apply to code snippets

#### Where should the reviewer start?

- Screenshots/run locally to look at new styles. Header, nav, tables, code snippets.

#### How should this be manually tested?

- Click around, see how you like interacting with the nav bar. Does it bother you that when you click a menu item, if a different menu item was previously selected, it's sub-items disappear? (Try clicking tokens, then elements)

#### FYI/Context

- To handle the nav bar subitem situation, I decided to use that `if` in liquid rather than CSS because the CSS solution got hairy and required way more code that felt less straightforward.
 
#### What are the relevant tickets?

https://3.basecamp.com/3494409/buckets/19192671/todolists/3628883354

#### Screenshots (if appropriate)

This screenshot shows new header, side-nav bar with a sub-item available under elements, and the rouge integrated:

Screen Shot 2021-04-07 at 7 09 28 AM

This screenshot shows updated tables that are more consistent in widths of cells:

Screen Shot 2021-04-07 at 7 10 23 AM

This one shows three table on tokens page, all with the same widths:

Screen Shot 2021-04-07 at 7 11 04 AM

#### Any other deploy steps?

No

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -54,6 +54,12 @@ collections:
 sass:
   sass_dir: css
 
+# syntax highlighting
+markdown: kramdown
```
Comment (ameseee) [inline _config.yml:58]: Do I need to specify a version >= 2.3.1 here? Not sure if I'm reading this correctly:

image

Comment (tannerwelsh) [reply _config.yml:58]: Shouldn't need to go in our `_config`, I think the way to update kramdown would be to use `bundle update kramdown`. So long as that moves it above 2.3.1 in our Gemfile.lock, we should be OK

Comment (ameseee) [reply _config.yml:58]: Ok good to know, thanks! I will do that.

**Comment by tannerwelsh** [comment]
So Amy I found a more direct way to get syntax highlighting working:

- No changes to `_config.yml`
- No changes from HTML to MD
- Just need to include CSS for the `.highlight` styles (like you did in this PR)
- And then to highlight code you just surround it with `{% highlight <language> %}` and `{% endhighlight %}` liquid tags, e.g.

```html
<p>Use an input element with a type attribute:
</p>
{% highlight html %}
  <input type="button" value="Submit">
{% endhighlight %}
```

Or when rendering HTML from our CSS docs' example:

```html
  {% assign buttons = site.elements | where: "element_type", "button" %}
  {% for button in buttons %}
    <tr class="token-table-row">
      <td class="element-code-cell">
        <code>{{ button.value }}</code>
      </td>
      <td>
        {% highlight html %}{{ button | remove: '<p>' | remove: '</p>' }}{% endhighlight %}
      </td>
```

To make this look ok, I also removed any existing CSS styles for `code` or `xmp` elements (which we probably shouldn't use anyway, since `xmp` is deprecated). Also tried rendering a multi-line example:

Screen Shot 2021-04-06 at 2 23 07 PM
Screen Shot 2021-04-06 at 2 23 05 PM

**Comment by tannerwelsh** [comment]
FWIW I was just going off of [these docs](https://jekyllrb.com/docs/liquid/tags/#code-snippet-highlighting) and [this post](https://bnhr.xyz/2017/03/25/add-syntax-highlighting-to-your-jekyll-site-with-rouge.html)

**Comment by ameseee** [comment]
This is so much better, thank you!

File: elements.html [hunk 2]
```diff
@@ -2,52 +2,69 @@
...
 permalink: /elements/
+subnav:
+  buttons-links: Buttons & Links
+  paragraphs: Paragraph Text
 ---
 
-<h2 class="s-h2">Buttons & Links</h2>
-<p>Many of our sites have links that appear to be buttons. This is acceptable from a design standpoint, but it is important that the HTML correctly communicates the intent of the element to the browser. Start by understanding the important difference between a button and a link:</p>
+<a id="buttons-links"></a>
+<h2 class="s-h2">Buttons & Links</h2></a>
```
Comment (tannerwelsh) [inline elements.html:11]: Is there a specific reason for using a separate `<a>` tag here? Could we just add the `id` to `h2`?

```suggestion
<h2 id="buttons-links" class="s-h2">Buttons & Links</h2></a>
```

File: elements.html [hunk 3]
```diff
@@ -87,23 +98,19 @@ <h3 class="s-h3">Savile Links</h3>
...
       <td class="element-sample-cell">{{ link }}</td>
     </tr>
     {% endfor %}
   </tbody>
 </table>
 
-<h2 class="s-h2">Paragraph Text</h2>
+<a id="paragraphs">
+  <h2 class="s-h2">Paragraphs</h2>
+</a>
```
Comment (tannerwelsh) [inline elements.html:113]: Same as above - can we just use `id` on the `h2` element and avoid wrapping with `<a>`?

File: elements.html [hunk 4]
```diff
@@ -118,36 +125,18 @@ <h2 class="s-h2">Paragraph Text</h2>
...
-      <td>
-        <code class="language-html" data-lang="html">
-          {{text | remove: '<p>' | remove: '</p>' | escape}}
-        </code>
-      </td>
-      </td>
-      <td class="element-sample-cell">
-        {{ text }}
-      </td>
+    <tr class="table-row">
```
Comment (tannerwelsh) [inline elements.html:128]: All these repeated tables are making me wonder if it's time to start abstracting them out into [Jekyll includes](https://jekyllrb.com/docs/includes/)

File: tokens.html [hunk 5]
```diff
@@ -2,27 +2,31 @@
 title: Tokens
 layout: default
 permalink: /tokens/
+subnav:
+  font-size: font-size
```
Comment (tannerwelsh) [inline tokens.html:6]: Can we use capitalized versions for the visible sub nav links?
```suggestion
  font-size: Font Size
```

File: tokens.html [hunk 6]
```diff
@@ -2,27 +2,31 @@
...
+  spacing: spacing
+  font-weight: font-weight
+  font-family: font-family
 ---
 
 {% assign grouped_tokens = site.tokens | group_by: "token_type" %}
 
 {% for token_group in grouped_tokens %}
-  <h2>{{ token_group.name | capitalize }}</h2>
+  <a id="{{token_group.name}}"><h2>{{ token_group.name | capitalize }}</h2></a>
```
Comment (tannerwelsh) [inline tokens.html:16]: Same question as above re: using `id` in `<a>` tags

**Review by tannerwelsh — CHANGES_REQUESTED** [review]
Overall I think the navbar is working great! And the syntax highlighting is 🔥 . A few small questions & suggestions added inline.

File: _layouts/default.html [hunk 7]
```diff
@@ -3,7 +3,8 @@
   <head>
     <meta charset="utf-8">
     <title>{{ page.title }}</title>
-    <!-- changed paths below bc I'm deploying differently from real Savile -->
+    <link rel="shortcut icon" type="image/png" href="{{site.assets_url}}/turing-school-mark-256.png" />
```
[the remark this answers is not in the document]
Comment (ameseee) [reply _layouts/default.html:6]: Oh thank you! I think I found it... somewhere, and didn't dig into researching it enough. Do we care to keep the comments that label what each link does, or remove?

File: css/syntax.css [hunk 8]
```diff
@@ -0,0 +1,82 @@
+code {
```
[the remark this answers is not in the document]
Comment (ameseee) [reply css/syntax.css:1]: That's a great question you ask. I snagged this from the FE/BE curriculum sites thinking it makes sense to match what they use... and there was no indication on those where it came from😅 I will poke around and see if I can backwards find it.

File: elements.html [hunk 9]
```diff
@@ -2,52 +2,69 @@
...
+  paragraphs: Paragraph Text
 ---
 
-<h2 class="s-h2">Buttons & Links</h2>
-<p>Many of our sites have links that appear to be buttons. This is acceptable from a design standpoint, but it is important that the HTML correctly communicates the intent of the element to the browser. Start by understanding the important difference between a button and a link:</p>
+<a id="buttons-links"></a>
+<h2 class="s-h2">Buttons & Links</h2></a>
+<p>Many of our sites have links that appear to be buttons. This is acceptable from a design standpoint, but it is
+  important that the HTML correctly communicates the intent of the element to the browser. Start by understanding the
+  important difference between a button and a link:</p>
```
[the remark this answers is not in the document]
Comment (ameseee) [reply elements.html:14]: GR I have no idea how this is happening. That was not on purpose! I wonder if it's when I did the change to markdown then back to html? I went through a whole rebase situation. Anyways, I'm changing it to one line.

File: elements.html [hunk 10]
```diff
@@ -2,52 +2,69 @@
...
-<xmp>
-  <a href="https://apply.turing.io" class="[add class(es) to make this appear as a button]">
-    <span>Apply Now</span>
-  </a>
-</xmp>
-<xmp>
-  <a href="/programs" class="[add class(es) to make this appear as a button]">More Info</a>
-</xmp>
+<p>In the following examples, the button-look-alike is actually an anchor tag. This is the correct tag to use because
+  the purpose of the element is to allow the user to navigate to a new page.</p>
```
[the remark this answers is not in the document]
Comment (ameseee) [reply elements.html:54]: ooh bc of the line break situation again, I bet? Thank you.

File: elements.html (hunk 4 above)
Comment (ameseee) [reply elements.html:112]: Yes, me too. At least for the ones that are the same on each page, like all the tokens, etc.

File: _layouts/default.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (tannerwelsh) [reply _layouts/default.html]: 🤷🏼  I think either way is fine. They aren't particularly helpful given that the file names are pretty self-explanatory.

File: css/syntax.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (tannerwelsh) [reply css/syntax.css:1]: Ah no worries :) I think so long as we reference that it came from the curriculum sites. :)

**Review by tannerwelsh — APPROVED** [review]

**Pull request merged**